File: leetcode-algorithms/043. Multiply Strings/solution.cpp
string multiply(string num1, string num2) {
    auto l = num1.size() + num2.size();

    vector<int> v(l,0);

    for(int i=num1.size()-1;i>=0;i--)
        for(int j = num2.size()-1;j>=0;j--){
            int k = (num1[i]-'0') * (num2[j]-'0');
            int p1 = i+j;
            int p2 = i+j+1;
            int sum = k + v[p2];
            v[p1] += sum / 10;
            v[p2] = sum %10;
    }



    string s="";

    for(int k:v){
        if((s.size()==0 && k==0))
            continue;
        s.push_back(k+'0');
    }


    return s.size()==0?"0":s;
}
int main()
{
	return 0;
}

File: leetcode-algorithms/043. Multiply Strings/solution.py
def multiply(num1, num2):
	"""
	:type num1: str
	:type num2: str
	:rtype: str
	"""
	
	l = len(num1) + len(num2)
	res = [0]*l


	for i in range(len(num1)-1,-1,-1):
		for j in range(len(num2)-1,-1,-1):
			k = int(num1[i]) * int(num2[j])
			p1, p2 = i+j, i+j+1

			s = k + res[p2]

			res[p1] += s // 10
			res[p2] = s % 10

	s = ''

	for i in res:
		if not (i ==0 and len(s)==0):
			s += str(i)

	return s if len(s)!=0 else '0'




if __name__ == '__main__':
	res = multiply('123','45')
	print(res)